File: workout-pekko/src/main/scala/workout/domain/CreateWorkoutRequest.scala
package workout.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto._

case class CreateWorkoutRequest(
    name: String,
    description: Option[String],
    workoutType: WorkoutType,
    durationMinutes: Int,
    caloriesBurned: Option[Int],
    difficulty: Difficulty
)

object CreateWorkoutRequest {
  implicit val encoder: Encoder[CreateWorkoutRequest] = deriveEncoder[CreateWorkoutRequest]
  implicit val decoder: Decoder[CreateWorkoutRequest] = deriveDecoder[CreateWorkoutRequest]
}

File: workout-spring-webflux/src/main/java/com/workout/app/api/dto/CreateWorkoutRequest.java
package com.workout.app.api.dto;

import com.workout.app.domain.Difficulty;
import com.workout.app.domain.WorkoutType;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateWorkoutRequest(
        @NotBlank String name,
        String description,
        @NotNull WorkoutType workoutType,
        @Min(1) int durationMinutes,
        Integer caloriesBurned,
        @NotNull Difficulty difficulty
) {}

